### PR TITLE
[BDTT-37] Completed full integration of sign up process

### PR DESCRIPTION
**Pull request by JeffS97**
# Description

Fixes [BDTT-37](https://bottlenose.atlassian.net/browse/BDTT-37)

[screen-capture.webm](https://user-images.githubusercontent.com/40801372/214379392-b6b2e640-2a8d-450e-b557-c8315e493092.webm)

List all acceptance criterias of user story and indicate which has been solved previously and which is solved currently
- [x] Customer leaves any field blank
- [x] Customer enters invalid email address
- [x] Customer enters existing username
- [x] Customer enters a password less than 8 characters
- [x] Customer keys in valid inputs for all fields

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
# Checklist:

- [x] I have added the appropriate labels to my PR
- [x] My code follows the style guidelines of this project
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by charmjj — CHANGES_REQUESTED** [review]

File: app/src/components/signUp/phoneInput.jsx [hunk 1]
```diff
@@ -1,31 +1,24 @@
...
+              className='border border-blue-400 font-normal text-black text-sm rounded-lg focus:outline-blue-400 block w-full p-2.5'
+              placeholder='Contact Number'
+          />
+      </label>
+  )
 }
 
-export default PhoneNumberField;
\ No newline at end of file
+// singaporePhoneNumberRegex = /^[0-9]{8}$/;
```
[the remark this answers is not in the document]
Comment (JeffS97) [reply app/src/components/signUp/phoneInput.jsx:23]: Removed

File: app/src/components/signUp/index.jsx [hunk 2]
```diff
@@ -15,11 +17,55 @@ export default function SignUp() {
...
+            setError("Error: Password length too short");
+            return;
+        }
+
+        const res = await signUp(email, username, password, phone, company);
+        if (res.code === 201) {
+            toggleModal();
+            setError("");
+            console.log("success");
+            // TODO: Redirect to home page/dashboard
```
[the remark this answers is not in the document]
Comment (JeffS97) [reply app/src/components/signUp/index.jsx:56]: Added navigation to sign in page

File: app/src/components/signUp/index.jsx [hunk 3]
```diff
@@ -15,11 +17,55 @@ export default function SignUp() {
...
+        } else if (res.code !== 500) {
+            setError("User already exists in the system");
```
[the remark this answers is not in the document]
Comment (JeffS97) [reply app/src/components/signUp/index.jsx:58]: Changed

File: app/src/components/signUp/registerButton.jsx [hunk 4]
```diff
@@ -1,20 +1,12 @@
 import React, { useState } from "react";
-import RegisterModal from "./registerModal";
 
 function RegisterButton() {
```
[the remark this answers is not in the document]
Comment (JeffS97) [reply app/src/components/signUp/registerButton.jsx:3]: Refactor in future sprints would be better i think

File: app/src/components/signUp/index.jsx [hunk 5]
```diff
@@ -31,19 +77,21 @@ export default function SignUp() {
...
                     </div>
-                    <PasswordField />
+                    <PasswordField password={password} setPassword={setPassword} />
                     <div className="flex justify-end">
                         <RegisterButton />
                     </div>
                 </div>
-
+                </form>
+                <div className='pt-5 text-red-500'>{ error }</div>
```
[the remark this answers is not in the document]
Comment (JeffS97) [reply app/src/components/signUp/index.jsx:95]: Changed

File: app/src/components/signUp/index.jsx [hunk 6]
```diff
@@ -15,11 +17,55 @@ export default function SignUp() {
...
+            setError("Error: Email not in the right format");
+            return;
+        }
+
+        if (password.length < 8){ //AC 4
+            setError("Error: Password length too short");
+            return;
+        }
+
+        const res = await signUp(email, username, password, phone, company);
```
[the remark this answers is not in the document]
Comment (JeffS97) [reply app/src/components/signUp/index.jsx:51]: Changed

**Review by charmjj — APPROVED** [review]

**Pull request merged**